include/deal.II/grid/cell_id.h

```diff
@@ -158,6 +158,12 @@ class CellId
   void
   serialize(Archive &ar, const unsigned int version);
 
+  /**
+   * Return the id of the coarse cell.
+   */
+  unsigned int
+  get_coarse_cell_id() const;
```

Comment:
Are we okay of this being an int or do we ever expect to have more than 4b cells?

Reply:
I don't think you want to read a 4b cell mesh from disk (or any parallel file system) now or in the next ten years. @peterrum would there be any internal inefficiency to make it 64 bits in any case?

Reply:
So: I modified the type of `cell_id` to `unsigned long int` (and I have introduced `numbers::invalid_unsigned_long_int`). Is that fine or should I use some deal.II alias?

It is nice that you are exited about `p:f:t` (which hopefully can handle so big meshes) and you are thinking about the long term future (Agenda 2030), but my short term goal is that @tamiko is happy and he is happy if he can run simulations on meshes with more than 10 millions coarse cells efficiently ;)

Note: in an upcoming PR we have to update the binary representation of `CellId` such that it can handle `unsigned long int` (and it is easily modifiable with simple C-shifts and -logical operations!!!).

Reply:
Furthermore, we might need to use other data structures in the DoFHandlerPolicy. Currently it relies on the quadrant the p4est quadrants...